microsoft/testsuites/dpdk/dpdksuite.py

```diff
@@ -55,10 +67,55 @@ def check_dpdk_build(self, node: Node, log: Logger) -> None:
             f"TX-PPS ({tx_pps}) should have been greater than 2^20 (~1m) PPS."
         ).is_greater_than(2 ** 20)
 
-    def _execute_expect_zero(self, node: Node, cmd: str) -> str:
-        result = node.execute(cmd, sudo=True, shell=True)
-        result.assert_exit_code()
-        return result.stdout
+    @TestCaseMetadata(
+        description="""
+            Tests a basic sender/receiver setup.
+        """,
+        priority=2,
+        requirement=simple_requirement(
+            min_nic_count=2,
+            network_interface=Sriov,
+            min_count=2,
+        ),
+    )
+    def check_2node_snd_rcv(self, environment: Environment, log: Logger) -> None:
+        test_kits = _init_nodes_concurrent(environment, log)
+        sender, receiver = test_kits
+
+        # helpful to have the public ip for debugging
+        external_ips = [
+            x.node.execute("curl ifconfig.me").stdout for x in [sender, receiver]
```

Comment:
The IP can get from connection info of node. It doesn't need to get from outside.

Reply:
what is the connection info get method? I don't see a parameter in Node for that

Reply:
It exists in RemoteNode. You can check it by `isinstance(node, RemoteNode)`. If it's LocalNode, skip this test case. After that, try `node.connection_info` get the information.

Reply:
thanks!